build.gradle

```diff
@@ -1588,7 +1588,6 @@ project(':test-common') {
     implementation project(':storage')
     implementation project(':server-common')
     implementation libs.slf4jApi
```

Comment:
Could you please take a look at @ijuma's comment (https://github.com/apache/kafka/pull/17373#discussion_r1894553658) 

maybe we can create a variable to collect them and then add them to implementation for each module:
```
  log4jRuntimeLibs = [
    libs.log4j1Bridge2Api
    libs.jacksonDatabindYaml
  ]
  log4jLibs = [
    libs.slf4jApi
    libs.slf4jLog4j2
    libs.log4j2Api
    libs.log4j2Core
  ]
...
implementation log4jLibs
runtimeOnly log4jRuntimeLibs
```

Reply:
I have tried it on my local machine but it led to tons of failure of tests.
I would keep debugging on my local machine.